backend/lcfs/web/api/transaction/repo.py

```diff
@@ -23,46 +29,69 @@ def __init__(self, db: AsyncSession = Depends(get_async_db_session)):
     async def get_transactions_paginated(self, offset: int, limit: int, conditions: list, sort_orders: list, organization_id: int = None):
         """
-        Fetch paginated, filtered, and sorted transactions.
+        Fetches paginated, filtered, and sorted transactions.
+        Adjusts visibility based on the requester's role: transferor, transferee, or government.
         
         Args:
             offset (int): Pagination offset.
             limit (int): Pagination limit.
             conditions (list): Filtering conditions.
             sort_orders (list): Sorting orders.
+            organization_id (int, optional): ID of the requesting organization; determines visibility rules. Defaults to government view if None.
         
         Returns:
             A tuple of (list of TransactionView instances, total count).
         """
-        if organization_id:
-            # Start with a condition to exclude "Draft" transactions, but include them if they match the specific criteria
-            base_condition = and_(
-                TransactionView.status != "Draft",
-                or_(
-                    TransactionView.from_organization_id == organization_id,
-                    TransactionView.to_organization_id == organization_id
-                )
-            )
+        query_conditions = []
+
+        # Base condition for transaction type "Transfer"
+        transfer_type_condition = TransactionView.transaction_type == 'Transfer'
 
-            # Include "Draft" transactions if organization_id matches from_organization_id
-            draft_condition = and_(
+        if organization_id is not None:
+            # Fetch visible statuses for both roles for "Transfer" transactions
+            visible_statuses_transferor = await self.get_visible_statuses(EntityType.Transferor)
+            visible_statuses_transferee = await self.get_visible_statuses(EntityType.Transferee)
+
+            # Construct role-specific conditions, applying them only to "Transfer" transactions
+            transferor_condition = and_(
+                transfer_type_condition,
                 TransactionView.from_organization_id == organization_id,
-                TransactionView.status == "Draft"
+                TransactionView.status.in_([status.value for status in visible_statuses_transferor])
             )
-            
-            # Combine the base condition with the draft inclusion condition
-            combined_condition = or_(base_condition, draft_condition)
-            conditions.append(combined_condition)
+            transferee_condition = and_(
+                transfer_type_condition,
+                TransactionView.to_organization_id == organization_id,
+                TransactionView.status.in_([status.value for status in visible_statuses_transferee])
+            )
+
+            # For transactions that are not of type "Transfer", include them without visibility filtering
+            non_transfer_condition = and_(
```

Comment:
leave a TODO here because there will be additional visibility checks needed for other transaction types.

Reply:
The comment added. Thanks!